### DISCUSSION TITLE
Wait for v2?

### DISCUSSION THREAD

**Discussion by LouisDeGuinness** [flyerhq/flutter_chat_ui#566] [Q&A] [answered]
Hey,

i'm writing an app for a small group of people and want to implement a chat.

I'm not sure if i should wait for v2 or start right away (need to get it done asap).

Is there a ETA for v2? Or can i start using v1 and v2 is more like a drop-in replacement with better base, so i don't have to rewrite my code too when v2 is released.

Thank you,
Marc

**Comment by demchenkoalex** [comment]
Hi!

I am taking my time with `v2` since its core has to be extendable and customisable as much as possible and work for all chat use cases like gpt, api, firebase etc. Currently I am testing my solution against those examples. That being said, `v2` does not even have an input field, I just add messages without any styling to check if core of the package works correctly - so it still needs some time to be polished. Sometimes I meet blockers and can spend weeks trying to solve them, so it is hard to give you an ETA because of that. I could say it will be done next week and then tomorrow I will realise gpt does not work with the code I created and I have re-do something and be stuck another 2 weeks.

`v1` and `v2` will be totally incompatible, because I just removed `v1` code and started from scratch to improve performance, quality and UX of the chat, trying to include 2+ years of feedback into the core of the `v2`.

I would say that if you feel like `v1` will work for your use case without much code/modifications you could use that (since you will spend like a day setting it up) and then you can do another day when `v2` is released - or you could consider alternative packages and/or write from scratch, that would be up to you.

Unfortunately, this package is in the transition state right now and since you need it to work asap that all options I could come up with.

**Comment by demchenkoalex** [comment, marked as answer]
v2 dev build is available https://github.com/flyerhq/flutter_chat_ui/issues/646
